addons/mail/static/src/models/chat_window.js

```diff
@@ -104,6 +104,16 @@ registerModel({
             const lastVisible = this.manager.lastVisible;
             this.manager.swap(this, lastVisible);
         },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onClickClose(ev) {
+            ev.stopPropagation();
+            if (!this.exists()) {
```

Comment:
Do we really need this inside a `onClick` inside a model ?

Reply:
Yes, we cannot guarantee that invoking `t-on` in OWL components only happens on existing records.
It's possible that record is itself deleted, so we need to guard them.

Reply:
In my understanding, it's only valid for even that have a chance to trigger this guard (e.g. on-scroll)
If not, we need to do it consistently on every `onClick` (https://github.com/odoo/odoo/pull/88231/files#diff-564a4336843bb910f681be30591a6113725c8c61c802683e65833cdbc2779109R107).

Reply:
Yes, I think all `t-on` are affected by the problem. The ones that are not guarded do work currently, but are prone to break with small changes to the code.

Reply:
All of them potentially have the issue, but the probability of the issue on click is so low (clicking exactly before the animation frame of the delete) that we never really bothered with them, and we've never seen or heard about the issue on them in 2  years.

Reply:
I would not introduce extra guards on handlers, but I would keep those that we have (it's likely that the issue was manifesting on them to make us add it at some point).

Reply:
I don't mind introducing a "handler" category to have guard on all of them.
Just for short/mid term, to hide technical problem and have ease on mind for at least this problem until we have overall improved code that won't need this eventually. It's only a problem due to integration with OWL components and lack of maturity to model.js code.

Personally I understand we need the guard, but it's annoying to make changes and have a test failing and reminding me of this subtle technical detail that we have to deal with.